### ISSUE TITLE
Whether it is possible to make for forms a feature of updating of model at the expense of processing throttle?

### ISSUE THREAD

**Issue by splincode** [angular/angular#21612]
It was like this:

`<input  [(ngModel)]="firstName" [ngModelOptions]="{ updateOn: 'blur' }">`

Then new feature 

`<input [(ngModel)]="firstName" [ngModelOptions]="{ updateOn: 'throttle', limit: 1000 }">`

Update (async) the model only after the user has stopped printing and has passed 1 second. And usage:

```html
<input 
    [(ngModel)]="firstName" 
    [ngModelOptions]="{ updateOn: 'throttle', limit: 1000, updateOnChange: query($event) }">`
```

Or:

```html
<form [ngFormOptions]="{ updateOn: 'throttle', limit: 1000 }">
 
   <input [(ngModel)]="firstName" [ngModelOptions]="{ updateOnChange: query($event) }">
   <!-- or a simplified version -->
   <input [(ngModel)]="secondName" (change)="query($event)">

</form>
```

```typescript
class Component {
  
  public query(firstName: string) {
    // .. processing
  }

}
```

It will be better in performance than always detecting

```html
<input [(ngModel)]="firstName" (change)="query($event)">`
```

Thus, developers will not need to write each time such functions themselves (throttle)

**Comment by CDDelta** [comment]
If you want to process an input change after one second, you can use the rxjs debounce operator. If this were to be integrated into Angular, it would probably have a similar mechanism, so there would not be much performance improvement.

**Comment by trotyl** [comment]
Duplicate of #19705

**Comment by mhevery** [comment]
close as dup
